apps/rush-lib/src/api/RushConfiguration.ts

```diff
@@ -429,18 +429,218 @@ export type ResolutionStrategy = 'fewer-dependencies' | 'fast';
+
+    this._changesFolder = path.join(this._commonFolder, RushConstants.changeFilesFolderName);
+  }
+
+  /**
+   * Loads the configuration data from an Rush.json configuration file and returns
+   * an RushConfiguration object.
+   */
+  public static loadFromConfigurationFile(rushJsonFilename: string): MinimalRushConfiguration {
+    let resolvedRushJsonFilename: string = path.resolve(rushJsonFilename);
+    // Load the rush.json before we fix the casing. If the case is wrong on a case-sensitive filesystem,
+    // the next line show throw.
+    const rushConfigurationJson: IRushConfigurationJson = JsonFile.load(resolvedRushJsonFilename);
+
+    try {
+      resolvedRushJsonFilename = trueCasePathSync(resolvedRushJsonFilename);
+    } catch (error) {
+      /* ignore errors from true-case-path */
+    }
+
+    return new MinimalRushConfiguration(rushConfigurationJson, resolvedRushJsonFilename);
+  }
+
+  public static loadFromDefaultLocation(options?: ITryFindRushJsonLocationOptions): MinimalRushConfiguration {
+    const rushJsonLocation: string | undefined = MinimalRushConfiguration.tryFindRushJsonLocation(options);
+
+    if (rushJsonLocation) {
+      return MinimalRushConfiguration.loadFromConfigurationFile(rushJsonLocation);
+    } else {
+      throw Utilities.getRushConfigNotFoundError();
+    }
+  }
+
+  /**
+   * Find the rush.json location and return the path, or undefined if a rush.json can't be found.
+   */
+  public static tryFindRushJsonLocation(options?: ITryFindRushJsonLocationOptions): string | undefined {
+    const optionsIn: ITryFindRushJsonLocationOptions = options || {};
+    const verbose: boolean = optionsIn.showVerbose || false;
+    let currentFolder: string = optionsIn.startingFolder || process.cwd();
+
+    // Look upwards at parent folders until we find a folder containing rush.json
+    for (let i: number = 0; i < 10; ++i) {
+      const rushJsonFilename: string = path.join(currentFolder, 'rush.json');
+
+      if (FileSystem.exists(rushJsonFilename)) {
+        if (i > 0 && verbose) {
+          console.log('Found configuration in ' + rushJsonFilename);
+        }
+
+        if (verbose) {
+          console.log('');
+        }
+
+        return rushJsonFilename;
+      }
+
+      const parentFolder: string = path.dirname(currentFolder);
+      if (parentFolder === currentFolder) {
+        break;
+      }
+
+      currentFolder = parentFolder;
+    }
+
+    return undefined;
+  }
+
+  /**
+   * Gets the JSON data structure for the "rush.json" configuration file.
+   *
+   * @internal
+   */
+  public get rushConfigurationJson(): IRushConfigurationJson {
+    return this._rushConfigurationJson;
+  }
+
+  /**
+   * The version of rush specified by the rushVersion property of the rush.json configuration file. This
+   * should be a semver style version number like "4.0.0"
+   */
+  public get rushVersion(): string {
+    return this._rushConfigurationJson.rushVersion;
```

Comment:
Note: this has changed so it no longer supports the "rushMinimumVersion" property of Rush <= 2.x